src/utils/queue.ts

```diff
@@ -15,29 +13,28 @@ class Queue {
         this.concurrent = concurrent;
     }
 
-    public execute<T>(func: () => Promise<T>, key: string | number = null): Promise<T> {
-        return new Promise((resolve, reject): void =>  {
-            this.jobs.push({key, func, running: false, resolve, reject});
+    public async execute<T>(func: () => Promise<T>, key: string | number = null): Promise<T> {
```

Comment:
the change here isn't strictly necessary, but it makes for nicer async stack traces since the call to `func` is in the same async stack chain as the caller.

Reply:
Cool, thanks!